### PR TITLE
added settings for language control

### PR DESCRIPTION
**Pull request by sprakash57**
## PR Checklist

Please check all that apply to this PR using "x":

- [x] I have checked that this PR is not a duplicate of an existing PR (open, closed or merged)
- [x] I have checked that this PR does not introduce a breaking change
- [ ] This PR introduces breaking changes and I have provided a detailed explanation below

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting)
- [ ] Refactoring (no functional changes)
- [ ] Documentation changes
- [ ] Other - Please describe:

## Fixes

Issue Number: #76 #92 

## Other information
- Wrapped all strings inside the localization function.
- Created option to set default locale ( language ) and control on context state.
- Added messages.fr.json
- Added some missing fields for Portuguese and German locale.

### CODE REVIEW

File: src/components/toolbar/index.js [hunk 1]
```diff
@@ -117,13 +123,16 @@ class Toolbar extends Component {
...
+                fullscreen
+                  ? Localization['fullscreen.off']
+                  : Localization['fullscreen.on']
               }
               tooltipClass={'right-edge'}
               onClick={toggleFullscreen}
               disabled={renderError}
             />
           )}
+          <Select options={Object.keys(Localization._props)} />
```
Comment (btzr-io) [inline src/components/toolbar/index.js]: Not sure what is this for used :thinking:

Comment (btzr-io) [reply src/components/toolbar/index.js]: but I think it should use: `getAvailableLanguages()`

Comment (btzr-io) [reply src/components/toolbar/index.js]: Also probably just called once, not in update / render function.

[the remark this answers is not in the document]
Comment (btzr-io) [reply src/components/toolbar/index.js]: please fix this ^^

File: src/components/toolbar/select.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  }
+
+  handleClick = e => {
+    this.context.toggleLang(e.target.value)
+  }
+
+  render() {
+    const { options } = this.props
+    return (
+      <select onClick={this.handleClick}>
```
[the remark this answers is not in the document]
Comment (btzr-io) [reply src/components/toolbar/select.js]: please fix this ^^

File: src/components/toolbar/index.js (hunk 1 above)
Comment (sprakash57) [reply src/components/toolbar/index.js]: Sure... i am working on the changes.

File: src/components/toolbar/select.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  }
+
+  render() {
+    const { options, icon } = this.props
+    return (
+      <>
+        <label style={{ marginTop: '5px', marginRight: '5px' }}>
+          <Icon path={icon} size={'22px'} className={'villain-icon'} />
+        </label>
+        <select onClick={this.handleClick}>
```
[the remark this answers is not in the document]
Comment (sprakash57) [reply src/components/toolbar/select.js]: Added label as icon.

File: src/components/toolbar/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (sprakash57) [reply src/components/toolbar/index.js]: Renamed component to avoid any confusion.

Comment (sprakash57) [reply src/components/toolbar/index.js]: I have renamed all the keys according to their English equivalent value. Also thanks for your suggestion i didn't know about `getAvailableLanguages()`  earlier.

File: src/components/toolbar/select.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (btzr-io) [reply src/components/toolbar/select.js]: it needs a `for` attribute or you can just wrap the select component inside the label.

I think better to just wrap it ^^

[the remark this answers is not in the document]
Comment (sprakash57) [reply src/components/toolbar/select.js]: Wrapped under label tag. Also i have capitalized locale code for better visibility.

**Comment by btzr-io** [comment]
@sprakash57 Nice great contribution, probably I'll end up removing the select component from the toolbar itself, because there is no  more space on the toolbar. see: #13 #34

**Comment by btzr-io** [comment]
@sprakash57 Does the auto language detection works ?

**Comment by btzr-io** [comment]
> The first language is considered the default one, so if a translation is missing for the selected language, the default one is shown and a line is written to the log as a reminder.

File: src/components/toolbar/zoom.js [hunk 4]
```diff
@@ -75,7 +75,7 @@ class ZoomControls extends Component {
         <Button
           type={'icon'}
           icon={mdiMinus}
-          tooltip={messages['zoom.out']}
+          tooltip={messages['Zoom out']}
```
Comment (btzr-io) [inline src/components/toolbar/zoom.js]: this should use the new `Localize` implementation

File: src/components/toolbar/zoom.js [hunk 5]
```diff
@@ -66,7 +66,7 @@ class ZoomControls extends Component {
       <React.Fragment>
         <Button
           type={'icon'}
-          tooltip={messages['zoom.in']}
+          tooltip={messages['Zoom in']}
```
Comment (btzr-io) [inline src/components/toolbar/zoom.js]: this should use the new `Localize` implementation

File: src/context.js [hunk 6]
```diff
@@ -23,6 +24,7 @@ const defaultState = {
   fullscreen: false,
   showControls: false,
   autoHideControls: false,
+  changeLang: Localization.getLanguage(),
```
Comment (btzr-io) [inline src/context.js]: this should use `localization.getInterfaceLanguage()` also rename to `language`

File: src/context.js [hunk 7]
```diff
@@ -68,6 +70,11 @@ export class ReaderProvider extends Component {
     }))
   }
 
+  toggleLang = lang => {
```
Comment (btzr-io) [inline src/context.js]: rename to `toggleLanguage` for readability

**Comment by btzr-io** [comment]
@sprakash57 Can you add language `auto-detection` ?

**Comment by btzr-io** [comment]
can you fix the conflicts ?

**Pull request merged**